mcs/class/System.Windows.Forms/System.Windows.Forms/XplatUIWin32.cs

```diff
@@ -890,7 +890,7 @@ private string RegisterWindowClass (int classStyle)
 				if (class_name != null)
 					return class_name;
 
-				class_name = string.Format ("Mono.WinForms.{0}.{1}", System.Threading.Thread.GetDomainID ().ToString (), classStyle);
+				class_name = string.Format ("Mono.WinForms.{0}.{1}.{2}", System.Threading.Thread.GetDomainID ().ToString (), classStyle, new Random().Next());
```

Comment:
Is this preferable to unregistering with `UnregisterClass`?

This seems like it could still fail (randomly :-) ). Should we try the registration in a limited size loop?